core/src/Kokkos_Parallel_Reduce.hpp

```diff
@@ -75,7 +75,11 @@ struct Sum {
 
  public:
   KOKKOS_INLINE_FUNCTION
-  Sum(value_type& value_) : value(&value_), references_scalar_v(true) {}
+  Sum(value_type& value_) : value(&value_), references_scalar_v(true) {
+    static_assert(
+        SpaceAccessibility<typename Space::memory_space, HostSpace>::assignable,
+        "Space must be assignable from HostSpace!");
```

Comment:
I would prefer if the static assert was inside the class body (as a member declaration)